{"version":1,"ops":[{"type":1,"author":{"id":"ba525199d25430b51e28e20d937a0d7ca063a89c"},"timestamp":1427159171,"metadata":{"github-id":"MDU6SXNzdWU2Mzg3MDc3Ng==","github-url":"https://github.com/DFHack/stonesense/issues/35","origin":"github"},"title":"stonesense: sigsegv on exit","message":"When stonesense is running and DF is closed, I get a segmentation fault with the following backtrace:\n\n``` gdb\n#0  0xf70ae546 in pthread_mutex_lock () from /usr/lib32/libpthread.so.0\n#1  0xf71b96d2 in pthread_mutex_lock () from /usr/lib32/libc.so.6\n#2  0xf6d487ed in ?? () from /usr/lib32/libX11.so.6\n#3  0xf6d6321d in XrmDestroyDatabase () from /usr/lib32/libX11.so.6\n#4  0xf6d4a4d3 in _XFreeDisplayStructure () from /usr/lib32/libX11.so.6\n#5  0xf6d38249 in XCloseDisplay () from /usr/lib32/libX11.so.6\n#6  0xf78e46e9 in ?? () from /usr/lib32/libSDL-1.2.so.0\n#7  0xf78ed2c3 in ?? () from /usr/lib32/libSDL-1.2.so.0\n#8  0xf78dbea9 in SDL_VideoQuit () from /usr/lib32/libSDL-1.2.so.0\n#9  0xf78b36dd in SDL_QuitSubSystem () from /usr/lib32/libSDL-1.2.so.0\n#10 0xf78b3789 in SDL_Quit () from /usr/lib32/libSDL-1.2.so.0\n#11 0xf7deeb5f in SDL_Quit () from ./hack/libdfhack.so\n#12 0xf749473d in main (argc=1, argv=0xffffd914)\n    at /build/dwarffortress/src/dwarf_fortress_unfuck/g_src/enabler.cpp:797\n#13 0xf70db64e in __libc_start_main () from /usr/lib32/libc.so.6\n#14 0x0804c9a1 in ?? ()\n```\n\nI've been trying to recover a core file, but `ulimit -c 1` doesn't seem to have any effect. I am running libx11 1.6.2 and sdl 1.2.15.\n\nThis only happens when I run stonesense, so I am tentatively calling this a bug in stonesense. Anyway, see if you can reproduce it. All I had to do was\n1. Run dfhack.\n2. Run stonesense from the dfhack console while DF was still in the menu.\n3. Close DF by hitting 'Quit' in the menu.","files":null}]}